### PR TITLE
Makefile: Add missing uninstall target

### PR DESCRIPTION
**Pull request by mirsal**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This commit adds the uninstall Makefile target, which reverts the effect of install.

### CODE REVIEW

**Comment by yan12125** [comment]
Is there a use case? ```make install``` is designed to be used with package managers (apt, yum, homebrew, ...), and package managers already provide uninstall commands.

**Comment by mirsal** [comment]
Hello,

Yen Chi Hsuan:
> Is there a use case? ```make install``` is designed to be used with package managers (apt, yum, homebrew, ...), and package managers already provide uninstall commands.

Well, ```make install``` can also be run manually, in which case the
user has no way to reverse its effects besides tracking which files were
installed where.

-- 
mirsal

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,18 @@ install: youtube-dl youtube-dl.1 youtube-dl.bash-completion youtube-dl.zsh youtu
...
+	rm -f $(DESTDIR)$(BINDIR)/youtube-dl
+	rmdir -p --ignore-fail-on-non-empty $(DESTDIR)$(BINDIR) || true
+	rm -f $(DESTDIR)$(MANDIR)/man1/youtube-dl.1
+	rmdir -p --ignore-fail-on-non-empty $(DESTDIR)$(MANDIR)/man1 || true
+	rm -f $(DESTDIR)$(SYSCONFDIR)/bash_completion.d/youtube-dl
+	rmdir -p --ignore-fail-on-non-empty $(DESTDIR)$(SYSCONFDIR)/bash_completion.d || true
+	rm -f $(DESTDIR)$(SHAREDIR)/zsh/site-functions/_youtube-dl/_youtube-dl
+	rmdir -p --ignore-fail-on-non-empty $(DESTDIR)$(SHAREDIR)/zsh/site-functions || true
+	rm -f $(DESTDIR)$(SYSCONFDIR)/fish/completions/youtube-dl.fish/youtube-dl.fish
+	rmdir -p --ignore-fail-on-non-empty $(DESTDIR)$(SYSCONFDIR)/fish/completions || true
```
Comment (yan12125) [inline Makefile]: rmdir shouldn't fail silently if directories are not empty

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,18 @@ install: youtube-dl youtube-dl.1 youtube-dl.bash-completion youtube-dl.zsh youtu
 	install -d $(DESTDIR)$(SYSCONFDIR)/fish/completions
 	install -m 644 youtube-dl.fish $(DESTDIR)$(SYSCONFDIR)/fish/completions/youtube-dl.fish
 
+uninstall:
+	rm -f $(DESTDIR)$(BINDIR)/youtube-dl
```
Comment (yan12125) [inline Makefile]: Better to add -v so that people know what are deleted

**Comment by mirsal** [comment]
Agreed, see new patch :)

File: Makefile [hunk 3]
```diff
@@ -26,6 +26,18 @@ install: youtube-dl youtube-dl.1 youtube-dl.bash-completion youtube-dl.zsh youtu
...
+uninstall:
+	rm -v -f $(DESTDIR)$(BINDIR)/youtube-dl
+	rmdir -p $(DESTDIR)$(BINDIR) || true
+	rm -v -f $(DESTDIR)$(MANDIR)/man1/youtube-dl.1
+	rmdir -p $(DESTDIR)$(MANDIR)/man1 || true
+	rm -v -f $(DESTDIR)$(SYSCONFDIR)/bash_completion.d/youtube-dl
+	rmdir -p $(DESTDIR)$(SYSCONFDIR)/bash_completion.d || true
+	rm -v -f $(DESTDIR)$(SHAREDIR)/zsh/site-functions/_youtube-dl/_youtube-dl
+	rmdir -p $(DESTDIR)$(SHAREDIR)/zsh/site-functions || true
+	rm -v -f $(DESTDIR)$(SYSCONFDIR)/fish/completions/youtube-dl.fish/youtube-dl.fish
```
Comment (yan12125) [inline Makefile]: Wrong path here

File: Makefile [hunk 4]
```diff
@@ -26,6 +26,18 @@ install: youtube-dl youtube-dl.1 youtube-dl.bash-completion youtube-dl.zsh youtu
...
 	install -m 644 youtube-dl.fish $(DESTDIR)$(SYSCONFDIR)/fish/completions/youtube-dl.fish
 
+uninstall:
+	rm -v -f $(DESTDIR)$(BINDIR)/youtube-dl
+	rmdir -p $(DESTDIR)$(BINDIR) || true
+	rm -v -f $(DESTDIR)$(MANDIR)/man1/youtube-dl.1
+	rmdir -p $(DESTDIR)$(MANDIR)/man1 || true
+	rm -v -f $(DESTDIR)$(SYSCONFDIR)/bash_completion.d/youtube-dl
+	rmdir -p $(DESTDIR)$(SYSCONFDIR)/bash_completion.d || true
+	rm -v -f $(DESTDIR)$(SHAREDIR)/zsh/site-functions/_youtube-dl/_youtube-dl
```
Comment (yan12125) [inline Makefile]: Wrong path here

Comment (mirsal) [inline Makefile]: Fixed in 26940b9

File: Makefile (hunk 3 above)
Comment (mirsal) [inline Makefile]: Fixed in 26940b9

**Comment by yan12125** [comment]
Still broken. Please make sure that no files are left after install and then uninstall
```
$ make install DESTDIR=~/ytdl-test
install -d ~/ytdl-test/usr/local/bin
install -m 755 youtube-dl ~/ytdl-test/usr/local/bin
install -d ~/ytdl-test/usr/local/man/man1
install -m 644 youtube-dl.1 ~/ytdl-test/usr/local/man/man1
install -d ~/ytdl-test/etc/bash_completion.d
install -m 644 youtube-dl.bash-completion ~/ytdl-test/etc/bash_completion.d/youtube-dl
install -d ~/ytdl-test/usr/local/share/zsh/site-functions
install -m 644 youtube-dl.zsh ~/ytdl-test/usr/local/share/zsh/site-functions/_youtube-dl
install -d ~/ytdl-test/etc/fish/completions
install -m 644 youtube-dl.fish ~/ytdl-test/etc/fish/completions/youtube-dl.fish

$ make uninstall DESTDIR=~/ytdl-test
rm -v -f ~/ytdl-test/usr/local/bin/youtube-dl
removed '/home/yen/ytdl-test/usr/local/bin/youtube-dl'
rmdir -p ~/ytdl-test/usr/local/bin || true
rmdir: failed to remove directory '/home/yen/ytdl-test/usr/local': Directory not empty
rm -v -f ~/ytdl-test/usr/local/man/man1/youtube-dl.1
removed '/home/yen/ytdl-test/usr/local/man/man1/youtube-dl.1'
rmdir -p ~/ytdl-test/usr/local/man/man1 || true
rmdir: failed to remove directory '/home/yen/ytdl-test/usr/local': Directory not empty
rm -v -f ~/ytdl-test/etc/bash_completion.d/youtube-dl
removed '/home/yen/ytdl-test/etc/bash_completion.d/youtube-dl'
rmdir -p ~/ytdl-test/etc/bash_completion.d || true
rmdir: failed to remove directory '/home/yen/ytdl-test/etc': Directory not empty
rm -v -f ~/ytdl-test/usr/local/share/zsh/site-functions/_youtube-dl/youtube-dl.zsh
rmdir -p ~/ytdl-test/usr/local/share/zsh/site-functions/_youtube-dl || true
rmdir: failed to remove '/home/yen/ytdl-test/usr/local/share/zsh/site-functions/_youtube-dl': Not a directory
rm -v -f ~/ytdl-test/etc/fish/completions/youtube-dl.fish
removed '/home/yen/ytdl-test/etc/fish/completions/youtube-dl.fish'
rmdir -p ~/ytdl-test/etc/fish/completions || true
rmdir: failed to remove directory '/home/yen/ytdl-test': Directory not empty

$ find ~/ytdl-test        
/home/yen/ytdl-test
/home/yen/ytdl-test/usr
/home/yen/ytdl-test/usr/local
/home/yen/ytdl-test/usr/local/share
/home/yen/ytdl-test/usr/local/share/zsh
/home/yen/ytdl-test/usr/local/share/zsh/site-functions
/home/yen/ytdl-test/usr/local/share/zsh/site-functions/_youtube-dl
```

**Pull request closed without merging**